engine/uState.ml

```diff
@@ -236,22 +237,23 @@ let process_universe_constraints ctx cstrs =
       else 
           match cst with
           | ULe (l, r) ->
-            if UGraph.check_leq univs l r then
-              (* Keep Prop/Set <= var around if var might be instantiated by prop or set
-                 later. *)
-              match Universe.level l, Universe.level r with
-              | Some l, Some r ->
-                Constraint.add (l, Le, r) local
-              | _ -> local
-            else
-              begin match Universe.level r with
-              | None -> user_err Pp.(str "Algebraic universe on the right")
-              | Some r' ->
-                if Level.is_small r' then
+            begin match Univ.Universe.level r with
+            | None ->
+              if UGraph.check_leq univs l r then local
+              else user_err Pp.(str "Algebraic universe on the right")
+            | Some r' ->
+              if Level.is_small r' then
                   if not (Universe.is_levels l)
                   then
+                    if UGraph.check_leq univs l r then local
+                    else
                     raise (UniverseInconsistency (Le, l, r, None))
                   else
+                    if UGraph.check_leq univs l r then match Univ.Universe.level l with
+                    | Some l ->
+                      Univ.Constraint.add (l, Le, r') local
```

Comment:
This is an already-verified constraint where `r'` is small, I don't think we need to add it. (we want to add new constraints and constraints Set <(=) x)